src/DataCell.js

```diff
@@ -38,8 +39,14 @@ export default class DataCell extends PureComponent {
   }
 
   render() {
-    const {row, col, rowSpan, readOnly, colSpan, width, overflow, value, className, editing, selected, onMouseDown, onMouseOver, onDoubleClick, onContextMenu} = this.props;
+    const {
+      row, col, rowSpan, readOnly, colSpan, width, overflow, value, className,
+      editing, selected, onMouseDown, onMouseOver, onDoubleClick, onContextMenu,
+      extraAttributes
+    } = this.props;
     const style = { width };
+    const filteredExtraAttribs = filterCellExtraAttributes(extraAttributes);
```

Comment:
This is the same data that gets called for every cell. 

You are filtering on every render. So for example, if I have the following 
```javascript
<Datagrid
   data={...1000 elements }
   extraAttributes={{'data-hint': "valid", 'hint': "invalid"}}
/>
```
This would mean for every re-render (edits/selection/etc), the filtering would happen again. 
There is no way to specify that a specific cell takes the data-hint